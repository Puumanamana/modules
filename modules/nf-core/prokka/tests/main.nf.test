// nf-core modules test prokka
nextflow_process {

    name "Test Process PROKKA"
    script "../main.nf"
    process "PROKKA"

    tag "modules"
    tag "modules_nfcore"
    tag "prokka"

    test("prokka - fasta") {
        when {
            process {
                """
                input[0] = [ [ id:'test', single_end:false ], // meta map
                    file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
            ]
                input[1] = []
                input[2] = []
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.gff,
                    process.out.gbk,
                    process.out.fna,
                    process.out.faa,
                    process.out.ffn,
                    process.out.sqn,
                    process.out.fsa,
                    process.out.tbl,
                    process.out.err,
                    process.out.txt,
                    process.out.tsv,
                    process.out.versions
                ).match() }
            )
        }

    }

}
