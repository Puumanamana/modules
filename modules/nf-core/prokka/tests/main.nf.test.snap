{
    "prokka - fasta": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.gff:md5,5dbfb8fcf2db020564c16045976a0933"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.gbk:md5,ec3532d5346e6ae7e18b9a7a32f880f1"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.fna:md5,787307f29a263e5657cc276ebbf7e2b3"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.faa:md5,a4ceda83262b3c222a6b1f508fb9e24b"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.ffn:md5,80f474b5367b7ea5ed23791935f65e34"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.sqn:md5,6bc6b4f865dd5d27f863c53431f6e2b0"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.fsa:md5,71bbefcb7f12046bcd3263f58cfd5404"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.tbl:md5,d8f816a066ced94b62d9618b13fb8add"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.err:md5,b3daedc646fddd422824e2b3e5e9229d"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.txt:md5,b40e485ffc8eaf1feacf8d79d9751a33"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.tsv:md5,da7c720c3018c5081d6a70b517b7d450"
                ]
            ],
            [
                "versions.yml:md5,e83a22fe02167e290d90853b45650db9"
            ]
        ],
        "timestamp": "2023-11-28T15:14:52.296875"
    }
}